sp/src/game/server/logicentities.cpp

```diff
@@ -27,6 +27,126 @@
+			0x74,0x79,0x47,0x72,0x6f,0x75,0x70,0x2e,0x61,0x70,0x70,0x65,0x6e,0x64,0x28,0x20,0x65,0x6e,0x74,0x20,
+			0x29,0x3b,0x0d,0x0a,0x09,0x09,0x69,0x66,0x20,0x28,0x20,0x65,0x6e,0x74,0x20,0x21,0x3d,0x20,0x6e,0x75,
+			0x6c,0x6c,0x20,0x29,0x0d,0x0a,0x09,0x09,0x7b,0x0d,0x0a,0x09,0x09,0x09,0x65,0x6e,0x74,0x2e,0x56,0x61,
+			0x6c,0x69,0x64,0x61,0x74,0x65,0x53,0x63,0x72,0x69,0x70,0x74,0x53,0x63,0x6f,0x70,0x65,0x28,0x29,0x3b,
+			0x0d,0x0a,0x09,0x09,0x09,0x65,0x6e,0x74,0x2e,0x47,0x65,0x74,0x53,0x63,0x72,0x69,0x70,0x74,0x53,0x63,
+			0x6f,0x70,0x65,0x28,0x29,0x2e,0x45,0x6e,0x74,0x69,0x74,0x79,0x47,0x72,0x6f,0x75,0x70,0x20,0x3c,0x2d,
+			0x20,0x45,0x6e,0x74,0x69,0x74,0x79,0x47,0x72,0x6f,0x75,0x70,0x3b,0x0d,0x0a,0x09,0x09,0x7d,0x0d,0x0a,
+			0x09,0x7d,0x0d,0x0a,0x7d,0x0d,0x0a,0x00
+		};
+
+		int iLastMember;
+		for ( iLastMember = MAX_SCRIPT_GROUP - 1; iLastMember >= 0; iLastMember-- )
+		{
+			if ( m_iszGroupMembers[iLastMember] != NULL_STRING )
+			{
+				break;
+			}
+		}
+
+		if ( iLastMember >= 0 )
+		{
+			HSCRIPT hAddScript = g_pScriptVM->CompileScript( szAddCode );
+			if ( hAddScript )
+			{
+				ValidateScriptScope();
+				m_ScriptScope.Run( hAddScript );
+				HSCRIPT hAddFunc = m_ScriptScope.LookupFunction( "__AppendToScriptGroup" );
+				if ( hAddFunc )
+				{
+					for ( int i = 0; i <= iLastMember; i++ )
+					{
+						m_ScriptScope.Call( hAddFunc, NULL, STRING(m_iszGroupMembers[i]) );
+					}
+					g_pScriptVM->ReleaseFunction( hAddFunc );
+					m_ScriptScope.ClearValue( "__AppendToScriptGroup" );
+				}
+
+				g_pScriptVM->ReleaseScript( hAddScript );
+			}
+		}
+		BaseClass::RunVScripts();
+	}
+
+	string_t m_iszGroupMembers[MAX_SCRIPT_GROUP];
+
+};
+
+LINK_ENTITY_TO_CLASS( logic_script, CLogicScript );
+
+BEGIN_DATADESC( CLogicScript )
+	// Silence, Classcheck!
+	// DEFINE_ARRAY( m_iszGroupMembers, FIELD_STRING, MAX_NUM_TEMPLATES ),
+
+	DEFINE_KEYFIELD( m_iszGroupMembers[0], FIELD_STRING, "Group00"),
+	DEFINE_KEYFIELD( m_iszGroupMembers[1], FIELD_STRING, "Group01"),
+	DEFINE_KEYFIELD( m_iszGroupMembers[2], FIELD_STRING, "Group02"),
+	DEFINE_KEYFIELD( m_iszGroupMembers[3], FIELD_STRING, "Group03"),
+	DEFINE_KEYFIELD( m_iszGroupMembers[4], FIELD_STRING, "Group04"),
+	DEFINE_KEYFIELD( m_iszGroupMembers[5], FIELD_STRING, "Group05"),
+	DEFINE_KEYFIELD( m_iszGroupMembers[6], FIELD_STRING, "Group06"),
+	DEFINE_KEYFIELD( m_iszGroupMembers[7], FIELD_STRING, "Group07"),
+	DEFINE_KEYFIELD( m_iszGroupMembers[8], FIELD_STRING, "Group08"),
+	DEFINE_KEYFIELD( m_iszGroupMembers[9], FIELD_STRING, "Group09"),
+	DEFINE_KEYFIELD( m_iszGroupMembers[10], FIELD_STRING, "Group10"),
+	DEFINE_KEYFIELD( m_iszGroupMembers[11], FIELD_STRING, "Group11"),
+	DEFINE_KEYFIELD( m_iszGroupMembers[12], FIELD_STRING, "Group12"),
+	DEFINE_KEYFIELD( m_iszGroupMembers[13], FIELD_STRING, "Group13"),
+	DEFINE_KEYFIELD( m_iszGroupMembers[14], FIELD_STRING, "Group14"),
+	DEFINE_KEYFIELD( m_iszGroupMembers[15], FIELD_STRING, "Group16"),
```

Comment:
The field name Group16 seems out of sync should be Group15, its copy/paste from ASW, but possibly should be fixed, these don't appear in the ASW FGD so can't see if they got it right in there, however looking other vscript FGD's it uses Group15 not Group16

Reply:
Resolved in 3ad34f3